rfqmsg/messages.go

```diff
@@ -0,0 +1,58 @@
+package rfqmsg
+
+import (
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// ID is the identifier for a RFQ message.
+type ID [32]byte
+
+// SerialisedScid is a serialised short channel id (SCID).
+type SerialisedScid uint64
+
+// TapMessageTypeBaseOffset is the taproot-assets specific message type
+// identifier base offset. All tap messages will have a type identifier that is
+// greater than this value.
+//
+// This offset was chosen as the concatenation of the alphabetical index
+// positions of the letters "t" (20), "a" (1), and "p" (16).
+const TapMessageTypeBaseOffset = 20116 + uint32(lnwire.CustomTypeStart)
+
+const (
+	// MsgTypeRequest is the message type identifier for a quote request
+	// message.
+	MsgTypeRequest = TapMessageTypeBaseOffset + 0
+
+	// MsgTypeAccept is the message type identifier for a quote accept
+	// message.
+	MsgTypeAccept = TapMessageTypeBaseOffset + 1
+
+	// MsgTypeReject is the message type identifier for a quote
+	// reject message.
+	MsgTypeReject = TapMessageTypeBaseOffset + 2
+)
+
+// WireMessage is a struct that represents a general wire message.
+type WireMessage struct {
+	// Peer is the origin/destination peer for this message.
+	Peer route.Vertex
+
+	// MsgType is the protocol message type number.
+	MsgType uint32
+
+	// Data is the data exchanged.
+	Data []byte
+}
+
+// IncomingMsg is an interface that represents an inbound wire message
+// that has been received from a peer.
+type IncomingMsg interface {
```

Comment:
Empty interface? Will read on to see the usage...


May want to make this "sealed" so having a private function s.t new instances can only be defined in tis package.